recipes/msfragger/meta.yaml

```diff
@@ -0,0 +1,50 @@
+{% set version = "4.0" %}
+
+# This key was generated ONLY for the testing of this bioconda package.
+# Users must generate their own key by agreeing to the terms at https://msfragger.arsci.com/upgrader/MSFragger-LICENSE.pdf.
+{% set academic_use_only_key = "f28c4be2-759b0a11-6c1ea652-1240565a" %}
+
+package:
+  name: msfragger
+  version: {{ version }}
+
+build:
+  number: 0
+  noarch: generic
+  run_exports:
+    - {{ pin_subpackage('msfragger', max_pin="x.x") }}
+
+requirements:
+  host:
+    - python >3.9
+  run:
+    - openjdk >=11
+    - python >=3.9
+    - mono >=5, <6
+    - zlib >=1.2.13
+
+test:
+  commands:
+    - msfragger --key {{ academic_use_only_key }} -Xms512m -Xmx1g --help  2>&1 | grep 'MSFragger'
+
+about:
+  home: https://github.com/Nesvilab/MSFragger
+  license: Academic License (https://msfragger.arsci.com/upgrader/MSFragger-LICENSE.pdf)
```

Comment:
Wondering if such a license is compatible with bioconda. Would be the first recipe in bioconda with such a license.

Reply:
We've had discussions with the developers and their legal team that approved publishing this package, as well as IonQuant and FragPipe. As I mentioned in another comment, the goal is to make sure any user knows the software must be used in accordance with the license.

It definitely is nuanced, so if there's a detail you'd like addressed let me know.

Reply:
HI @bernt-matthias - I just check where we stand here. Is the Bioconda team able to revisit this license again? The developers in the https://github.com/Nesvilab/MSFragger/issues/51 discussion were consulted for this package.

Reply:
I will ping the bioconda channel. But IMO the following is problematic

> LICENSEE agrees that it shall use the PROGRAM only for LICENSEE'S sole and exclusive use, and shall not disclose, sell, license, or otherwise distribute the PROGRAM, in whole or in part, to any third party without the prior written consent of MICHIGAN. 

- Bioconda will distribute this. 
- If this is installed on a public service, then its not controllable who uses it.

Reply:
I really would like to see MSFragger in bioconda, but I'm unsure if the license allows to ...

Reply:
I think this a showstopper, the license clearly doesn't allow redistribution.

Reply:
Hi @bernt-matthias and @nsoranzo,

Thanks for taking a look at this. I won't speak for them, but I think a written document for the Bioconda bioconda from the Michigan legal team can likely be produced, as this was previously done for MSFragger in Galaxy (@bgruening was a part of this original discussion). If I coordinate this, would that give you sufficient confidence to publish this tool?

Reply:
Thanks @reid-wagner , I've received the email from Tim with the support letter provided by the University of Michigan, so that looks resolved from my side.

Reply:
If @nsoranzo is fine with it I have no objections.

Still I have a few comments: 

- What happens if someone uses the `academic_use_only_key` given in `meta.yaml`? 
- The mechanism implemented here is IMO super annoying for downstream tools/projects using the software (OpenMS, Galaxy, ...), since they will need to adapt their CI to use such a key (which one?). Setting up CI and keep it running is already complicated enough. See [here](https://github.com/OpenMS/OpenMS/issues/7000#issuecomment-1670272055) for an example that this may be a reason to drop testing of the integration or the complete removal of the support for a software.

Reply:
Thanks @nsoranzo and @bernt-matthias!

> What happens if someone uses the academic_use_only_key given in meta.yaml?

That's an acceptable possibility. The argument is that we have given the user the opportunity to refrain from using the tool improperly. If a user seeks this key out, with this variable name and comment, then they must understand it's not for commercial use.

> The mechanism implemented here is IMO super annoying for downstream tools/projects using the software.

This was asked for by the Michigan team, but it can be bypassed in Galaxy tool. Again, it's more about inserting the legal agreements into a user's workflow rather than strict enforcement. The main class actually will run without this flag, but the wrapper script is what actually enforces supplying the flag. In Galaxy, there is required agreement clause. I can't speak to OpenMS - does the fact that it's non-secret, and will work indefinitely for a given version help?

Reply:
Thanks for the explanations. Makes more and more sense to me :)

Reply:
Sure!